File: teme_table.py
from sgp4.api import Satrec, WGS84
from sgp4.conveniences import sat_epoch_datetime,jday_datetime
from datetime import datetime, timedelta,timezone
import json

output_lines = []
with open("./2022_10_19.3le") as f:
    lines = [ x.strip() for x in f.readlines()]
    for i in range(0,len(lines),3):
        tle = [lines[i+1], lines[i+2]]
        satellite = Satrec.twoline2rv(*tle, WGS84)
        sat_epoch = sat_epoch_datetime(satellite)
        threshold = datetime(2022,10,18,tzinfo=timezone(timedelta()))
        t = datetime(2022,10,19,tzinfo=timezone(timedelta()))
        t_string = t.isoformat()

        if sat_epoch > threshold:
            jd,fr = jday_datetime(t)
            _, r, _ = satellite.sgp4(jd, fr)
            output_lines.append({"tle" : tle, "time" : t_string, "position" : [r[0], r[1], r[2]]})

with open("expected.json", "w") as g:
    json.dump(output_lines, g)

File: teme_table.js
const satellite = require('satellite.js');
const fs = require('fs');

let rawdata = fs.readFileSync('expected.json');
let student = JSON.parse(rawdata);
var t = new Date("2022-10-19T00:00:00+00:00");
console.log(t);

student.forEach(sat => {
    var satrec = satellite.twoline2satrec(sat.tle[0], sat.tle[1])
    var positionAndVelocity = satellite.propagate(satrec, new Date(sat.time));
    var dist = 1000 *Math.sqrt(Math.pow(sat.position[0] - positionAndVelocity.position.x,2) +
    Math.pow(sat.position[1] - positionAndVelocity.position.y,2) +
    Math.pow(sat.position[2] - positionAndVelocity.position.z,2) );
    if (dist > 1) {
        console.log(satrec.satnum,  dist);
    }
})
